template/interpolate/aws/secretsmanager/types.go

```diff
@@ -0,0 +1,36 @@
+package secretsmanager
+
+import (
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+)
+
+// AWSConfig store configuration used to initialize
+// secrets manager client.
+type AWSConfig struct {
+	Region string
```

Comment:
One thing I was thinking here is if we could use the config, possible session, from the packer/builder/amazon/common pkg so we don't have to maintain a separate config. But to be honest I don't what the change would entail off hand. I think this is something to look into as a separate PR.

Reply:
I thought exactly the same after seeing issue #9237. In my opinion it is not a bug but it is indeed confusing that interpolation function `aws_secretsmanager` cannot use AWS Builder configuration.

As you said it would be better to rework this in a different PR as I need to walk through Amazon Builder source code in order to better understand how to use Builder AWS client.